README.md

```diff
@@ -91,6 +92,31 @@ is provided by code that been generated from a model of the service.
 * [Contributions Guidelines](./documents/CONTRIBUTING.md)
 * [DEVELOPING](./documents/DEVELOPING.md)
 
+## FIPS Compliance
+
+This SDK supports [FIPS compliance](https://aws.amazon.com/compliance/fips/) on *some platforms*. If you're on a FIPS-compliant
+platform, FIPS will be enabled by default. See the [Platform-Specific JARs section](https://github.com/awslabs/aws-crt-java?tab=readme-ov-file#platform-specific-jars)
+in the CRT project for the current list of FIPS-compliant platforms.
+
+To interact with AWS IoT in a FIPS-compliant manner, you must use a [FIPS compliant endpoint](https://docs.aws.amazon.com/iot/latest/developerguide/iot-connect-fips.html#iot-connect-fips-data).
```

Comment:
Do we want add any test for the fips-compliant endpoint?

Reply:
I've added a sample connecting to the fips endpoint to CI